src/primitive/rectangular_plane.rs

```diff
@@ -0,0 +1,151 @@
+//! Rectangular plane primitive
+
+use cgmath::{BaseFloat, Point3, Vector2, Vector3};
+use cgmath::prelude::*;
+
+use {Aabb3, Ray3, Sphere};
+use prelude::*;
+use primitive::util::get_max_point;
+
+/// Rectangular plane primitive. Will lie on the xy plane when not transformed.
+///
+/// Have a cached set of corner points to speed up computation.
+#[derive(Debug, Clone, PartialEq)]
+#[cfg_attr(feature = "eders", derive(Serialize, Deserialize))]
+pub struct RectangularPlane<S> {
+    /// Dimensions of the rectangle
+    dim: Vector2<S>,
+    half_dim: Vector2<S>,
+    corners: Vec<Point3<S>>,
```

Comment:
any reason this is not `[Point3<S>; 4]`? Not sure if it's worth storing the corners at all, given how straightforward it is to produce them

Reply:
They are used atleast 6 times per intersection test, so it should be more performant to cache them